Source/msg.cpp

```diff
@@ -898,44 +900,47 @@ DWORD OnRespawnItem(const TCmd *pCmd, int pnum)
 
 DWORD OnAttackTile(const TCmd *pCmd, Player &player)
 {
-	auto *p = (TCmdLoc *)pCmd;
+	const auto &message = *reinterpret_cast<const TCmdLoc *>(pCmd);

```

Comment:
Can these `reinterpret_cast` result in reading out of bounds?
Example: A bad client sends a message with id `CMD_GOTOAGETITEM` but only send sizeof(TCmd) + 1 byte.
What will happen?
Bonus question: could this also occur in `HandleAllPackets`?

Reply:
Yes that seams likely. But it is only reads (`const`), and we are now validating that read content. So it wouldn't let you cause anything you couldn't simply do by providing a valid size and distinct values.

I'm unsure if `multi_process_network_packets()` currently does sufficient validation to prevent an OOB read in `HandleAllPackets()`, but `ProcessTmsgs()` should be using it safely at this point.

Reply:
If you read uncommitted memory couldn't you get a read access violation? And this could turn down the process?
I know this is unlikely and shouldn't be security issue cause at worst the app crashes, just wanted to mention it. 🙂

Reply:
Didn't mention it as it's unlikely :)

For 1.3.0 the main concern is security, but being able to ruin peoples fun isn't grate either. I have thought of ways to prevent it, but it would require some restructuring so I'm going to wait til after all the validation is done with seeing how i feel about it.